client/src/components/TagField.js

```diff
@@ -51,205 +73,155 @@ class TagField extends Component {
    * @param {string} input
    * @return {Promise<Array<Object>>|Promise<{options: Array<Object>}>}
    */
-  getOptions(input) {
-    const { lazyLoad, options } = this.props;
-
+  const getOptions = useCallback((input) => {
```

Comment:
Why are we using `useCallback` all over the place in this refactor? I don't think it was used in the previous two?

Reply:
As above can you please remove that hook?

Or alternatively should we agree that _all_ of the callbacks in functional components should use this callback and open a card to update the ones that aren't going through this process? (Or is there nuance there where some of them shouldn't use the hook?)

The latter will mean I need to do some reading up on how the hook works and what drawbacks or side effects it might have etc so will slow down review of at least this PR if not other refactor PRs - but if you think the optimisation is worth it I'm open to that.